chirp/drivers/radtel_rt620.py

```diff
@@ -0,0 +1,437 @@
+# Copyright 2024:
+# * Pavel Moravec, OK2MOP <moravecp.cz@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+from chirp import chirp_common, directory
+from chirp.settings import RadioSettingGroup, RadioSetting, \
+    RadioSettingValueBoolean, RadioSettingValueList
+from chirp.drivers import baofeng_uv17Pro, baofeng_uv17
+
+LOG = logging.getLogger(__name__)
+
+
+def _download_cal(radio):
+    if hasattr(radio, 'CAL_BLOCK') and \
+            radio.CAL_BLOCK not in radio.BLOCK_ORDER:
+        radio.BLOCK_ORDER.append(radio.CAL_BLOCK)
```

Comment:
This is not doing what (I think) you think it's doing. This is modifying the effectively-global definition of `UV17.BLOCK_ORDER`, as is the `.remove()` below(). That means if I do the following:

1. Download from an RT620 (`BLOCK_ORDER` gets modified)
2. Download from a UV17 in a new tab (modified `BLOCK_ORDER` gets used, may fail?)
3. Upload step 2 back to the UV17 (modified `BLOCK_ORDER` gets used, maybe cause damage?)

Things don't behave properly.

In order to make this correct, you need to store something on the object *instance* and not modify the mutable copy attached to the *class* like you're doing. The instance of the list on the UV17 is the same instance for it and all subclasses that do not override it.

Copying it and re-setting it on the radio instance here would be one way, but that's obscure and confusing. Probably better would be to make the `baofeng_uv17._download()` method take a parameter of `extra_blocks=...` or something.

It sounds like this is not actually implemented currently and we're never uploading the cal block right? If so, can we just remove that from this patch for the time being and work on it in a follow-on?

Reply:
You are probably right that it could generate issues for possible child classes. Again it was workaround for a single file driver, now I can patch it correctly. In the meantime I have found out the UV17 driver suffers same issue with Banks tab, so it needs to be patched anyway. So I will create 2 more PRs (one for UV-17Pro driver, one for UV-17 driver) and then rebase this PR on them

Reply:
It seems that my local copy of BLOCK_ORDER which should have prevented that got deleted some time during the edits and only the comment remained at that place. But it would be better to fix it by other means.

Reply:
This is a common thing in python that trips people up. Assignment is by reference. Consider this example:
```
class Foo:
    mylist = [1, 2, 3]

fooA = Foo()
fooB = Foo()

fooA.mylist.append(4)
print(fooB.mylist)
```
If you run that, it prints:
```
[1, 2, 3, 4]
```
Because there is only one copy of `mylist` and it's stored on the _class_, not copied for each instance. That's okay for global non-mutable stuff, but you have to be careful. This is why I say it would be better to just pass the block list to the download/upload function. But, if you're going your `BLOCK_O_READ` route, you need to be careful to copy it on the local object instance.

Reply:
I know it is a class variable, actually I am planning on using 
`BLOCK_O_READ = baofeng_uv17.UV17.BLOCK_O_READ + [2] # back up calibration`
This should be OK as it creates a new instance, right?

Reply:
Yes, that will do it :)